interceptors/validator/validator.go

```diff
@@ -83,9 +90,9 @@ func UnaryServerInterceptor(all bool) grpc.UnaryServerInterceptor {
 // returns ALL validation error as a wrapped multi-error.
 // Note that generated codes prior to protoc-gen-validate v0.6.0 do not provide an all-validation
 // interface. In this case the interceptor fallbacks to legacy validation and `all` is ignored.
-func UnaryClientInterceptor(all bool) grpc.UnaryClientInterceptor {
+func UnaryClientInterceptor(all bool, logger logging.Logger) grpc.UnaryClientInterceptor {
```

Comment:
I would definitely create optional argument here and everywhere else we use logger.

So users can do:
`UnaryClientInterceptor()`
`UnaryClientInterceptor(validator.WithLogger(...))`
`UnaryClientInterceptor(validator.WithFailFast())`

We can use this opportunity to clear out what all means (it's simply FailFast).

This is last moment before full v2.0 so we can (and should) break compatibility.

Would you be able to help with that? 🙈

Reply:
do we need to make similar changes for other interceptions too?

Reply:
Only validator interceptors (all interceptors in validator package)

Reply:
Otherwise the PR is perfect (: